protocol/tx/entry.go

```diff
@@ -26,10 +26,17 @@ type entry interface {
 var errInvalidValue = errors.New("invalid value")
 
 func entryID(e entry) (hash bc.Hash) {
+	// This nil test only handles the case where e is the zero value of
+	// the entry interface.
 	if e == nil {
 		return hash
 	}
 
+	// This nil test handles the case where e is a nil pointer.
+	if reflect.ValueOf(e).IsNil() {
```

Comment:
I preferred to leave it as two separate `if`s to avoid constructing a `reflect.Value` in the `e == nil` case, but otherwise took your advice.